.github/workflows/integrate.yaml

```
@@ -31,61 +33,22 @@ jobs:
       run: sudo apt install tox
 
     - name: Run unit tests
-      run: tox -e unit
+      run: |
+        tox -e tensorboard-controller-unit
+        tox -e tensorboards-web-app-unit
 
   deploy:
-    name: Test
+    name: Integration Test
     runs-on: ubuntu-latest
-
     steps:
-    - name: Check out repo
-      uses: actions/checkout@v2
-
-    - uses: balchua/microk8s-actions@v0.2.2
-      with:
-        addons: '["dns", "storage", "rbac"]'
-
-    - name: Install dependencies
-      run: |
-        set -eux
-        sudo pip3 install charmcraft
-        sudo snap install juju --classic
-        sudo snap install juju-helpers --classic
-        sudo snap install juju-wait --classic
-        sudo snap install yq
-
-    - name: Bootstrap Juju
-      run: |
-        set -eux
-        sg microk8s -c 'juju bootstrap microk8s uk8s'
-        juju add-model ci-test
-
-    - name: Deploy charm
-      run: |
-        set -eux
-        charmcraft build
-        juju deploy ./*.charm \
-          --resource oci-image=$(yq eval '.resources.oci-image.upstream-source' metadata.yaml)
-        juju deploy cs:istio-pilot
-        juju relate istio-pilot kubeflow-tensorboards
-        # https://bugs.launchpad.net/juju/+bug/1921739
-        (i=60; while ! juju wait -wvt 30 ; do ((--i)) || exit; sleep 1; done)
-
-    - name: Test charm
-      run: ls -lath
-
-    - name: Get all
-      run: kubectl get all -A
-      if: failure()
-
-    - name: Get juju status
-      run: juju status
-      if: failure()
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+            provider: microk8s
+      - name: Run test
+        run: |
+          tox -e integration
```

I like having the debugging logs here for failures.  Something like (note below was copied from a different app so it needs modification):

```suggestion
          tox -e integration

      # TODO: Remove once the actions-operator does this automatically
      - name: Configure kubectl
        run: |
          sg microk8s -c "microk8s config > ~/.kube/config" 
                             
      # On failure, capture debugging resources
      - name: Get all
        run: kubectl get all -A
        if: failure()

      - name: Describe deployments
        run: kubectl describe deployments -A
        if: failure()

      - name: Describe replicasets
        run: kubectl describe replicasets -A
        if: failure()

      - name: Get juju status
        run: juju status
        if: failure()

      - name: Get application logs
        run: kubectl logs -n testing --tail 1000 -ljuju-app=${{ matrix.charm }}
        if: failure()

      - name: Get argo-controller operator logs
        run: kubectl logs -n testing --tail 1000 -ljuju-operator=${{ matrix.charm }}
        if: failure()

```
```